backend/src/services/compilers/AlgoDebugCompilerImpl.ts

```diff
@@ -0,0 +1,43 @@
+import { CompilerRequest, CompilerResponse } from "../../types/compiler";
+import { OutputParser } from "../../utils/OutputParser";
+import { Compiler } from "./compilerFactory";
+
+export class AlgoDebugCompilerImpl implements Compiler {
+    static API_URL: string = "http://192.168.2.132:40042/compile";
```

Comment:
Wydaje mi się, że rozsądnie byłoby wrzucić ten URL (albo jego początek, IP) do .env

Reply:
Zrobiłem to na podobę tych połączeń z CodeX i JDoodle, że url połączenia do tych kompilatorów są podane w dedykowanych plikach, no bo inaczej to bym musiał dodać do .env 3 url mimo, że np w danym momencie używam jednego
No, ale z racji, że to jest nasz kompilator i może się tam jeszcze zmieni adres albo port to może jakiś sens to ma, żeby to wrzucić do env.
Moze jak ktoś jeszcze będzie robił CR to podzieli się swoim zdaniem

Reply:
skoro .env i tak jest publiczny a raczej nigdzie indziej ta ścieżka się nie pojawi to nie wiem czy jest sens

Reply:
wgl  ta cała klasa jest praktycznie taka sama jak CodexCompiler więc jakiś sens może to mieć jeśli by się tylko ścieżkę podmieniało ale znowu wtedy dziwnie to wyjdzie że JSDOODLE jest osobno

Reply:
No niby tak, klasy połączeń od CodeX i naszego kompilatora są bardzo podobne, ale gdybyśmy to zcalili to w przypadku jakiś zmian w jednym kompilatorze, trzeba by było znów to rozdzielać te klasy albo dodawać jakąś ifologię.

A co do url to ewentualnie mogę zrobić tak, że domyślnie bierze url z klasy, a jak jest podany adres w .env to go podmieniamy. Może mieć to jakieś zastosowanie, jakbyśmy postawili kilka instancji kompilatora na różnych adresach/portach i byśmy chcieli robić jakieś hard-testy, to żeby nie popsuć głównego kompilatora.